src/main/java/com/glencoesoftware/omero/zarr/ZarrPixelBuffer.java

```diff
@@ -825,26 +938,29 @@ public void setResolutionLevel(int resolutionLevel) {
 
             ZarrArray fullResolutionArray = zarrArrayCache.get(
                     root.resolve("0")).get();
-
-            // map each Z index in the full resolution array
-            // to a Z index in the subresolution array
-            // if no Z downsampling, this is just an identity map
-            int fullResZ = fullResolutionArray.getShape()[2];
-            int arrayZ = array.getShape()[2];
-            for (int z=0; z<fullResZ; z++) {
-                zIndexMap.put(z, Math.round(z * arrayZ / fullResZ));
+            
+            if (axesOrder.containsKey(Axis.Z)) {
+                // map each Z index in the full resolution array
+                // to a Z index in the subresolution array
+                // if no Z downsampling, this is just an identity map
+                int fullResZ = fullResolutionArray.getShape()[axesOrder.get(Axis.Z)];
+                int arrayZ = array.getShape()[axesOrder.get(Axis.Z)];
+                for (int z=0; z<fullResZ; z++) {
```

Comment:
Same formatting comment here

Reply:
That's true, I've not set any particular formatting style in my IDE, sorry. Happy to use the bioformats2raw one directly!